sqlmesh/core/config/connection.py

```diff
@@ -1276,3 +1274,16 @@ def _connection_config_validator(
     mode="before",
     check_fields=False,
 )(_connection_config_validator)
+
+
+if t.TYPE_CHECKING:
+    # TypeAlias hasn't been introduced until Python 3.10 which means that we can't use it
+    # outside the TYPE_CHECKING guard.
+    SerializableConnectionConfig: t.TypeAlias = ConnectionConfig  # type: ignore
+elif PYDANTIC_MAJOR_VERSION >= 2:
+    import pydantic
+
+    # Workaround for https://docs.pydantic.dev/latest/concepts/serialization/#serializing-with-duck-typing
+    SerializableConnectionConfig = pydantic.SerializeAsAny[ConnectionConfig]  # type: ignore
```

Comment:
Is this a pattern we'll need more? If so, consider wrapping the logic in a method provided by our pydantic util module.

Reply:
Don't want to make it generic until we actually need it. This code is trash and I'd rather keep it contained for now.